app/models/katello/content_view.rb

```diff
@@ -787,10 +787,29 @@ def composite_cv_components_changed?
       published_component_version_ids.compact.uniq.sort != unpublished_component_version_ids.compact.uniq.sort
     end
 
+    def last_publish_task_success?
+      last_publish_result = latest_version_object&.history&.publish&.first&.task&.result
+      return last_publish_result&.equal?('success')
+    end
+
     def needs_publish?
```

Comment:
```suggestion
    def needs_publish?
          # returns 
          # true - if audit records indicate changes since latest version
          # nil - if we have insufficient data 
         # false - otherwise
```

Reply:
format it correctly :) but theres  value in stating what the expected behavior is here

Reply:
Updates look good